### PR TITLE
Deprecate prep kwarg and remove third list from process_queue

### PR DESCRIPTION
**Pull request by lillian542**
The empty third list returned by `process_queue` is removed, and a deprecation warning is added when the `prep` kwarg is used in `QuantumScript`

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-43884]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -24,6 +24,10 @@
   
 <h3>Deprecations 👋</h3>
 
+* The ``prep`` keyword argument in ``QuantumScript`` is deprecated and will be removed from `QuantumScript`.
+  ``StatePrepBase`` operations should be placed at the beginning of the `ops` list instead.
+  [#4554](https://github.com/PennyLaneAI/pennylane/pull/4554)
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:29]:
```suggestion
  [(#4554)](https://github.com/PennyLaneAI/pennylane/pull/4554)
```

**Review by albi3ro — APPROVED** [review]

File: pennylane/queuing.py [hunk 2]
```diff
@@ -580,4 +578,4 @@ def process_queue(queue: AnnotatedQueue):
                 )
```
Comment (timmysilv) [inline pennylane/queuing.py:578]: the docstring should be updated here, effectively just doing the opposite of [this commit](https://github.com/PennyLaneAI/pennylane/commit/d4a500ae99070eed76658586372b7e1bf1dfc55b)

**Review by timmysilv — COMMENTED** [review]
please uncomment [this](https://github.com/PennyLaneAI/pennylane/blob/15bcf041a71effad70c3789d84b1b25505cf5c4d/tests/tape/test_qscript.py#L100-L101) `pytest.warns` and indent the line below it to catch this warning. One extra comment in-line, otherwise looks good!

**Pull request merged**